diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..4cfec06
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,7 @@
+project('InfiniTime', 'c',
+  version : '0.1',
+  default_options : ['warning_level=3'])
+
+# this is just to satisfy Meson until https://github.com/mesonbuild/meson/issues/12451 is fixed
+infinitime_dir = meson.project_source_root()
+freertos_config = files('src' / 'FreeRTOSConfig.h')
